google/resource_compute_router_interface.go

```diff
@@ -240,6 +240,10 @@ func resourceComputeRouterInterfaceDelete(d *schema.ResourceData, meta interface
 		Interfaces: newIfaces,
 	}
 
+	if len(newIfaces) == 0 {
+		patchRouter.ForceSendFields = append(patchRouter.ForceSendFields, "Interfaces")
```

Comment:
This is fine with me, though we probably don't even need the length check- we could just do:
```go
patchRouter := &compute.Router{
  Interfaces: newIfaces,
  ForceSendFields: []string{"Interfaces"}, // allows deleting the last one
}
```

up to you!

Reply:
I kind of prefer the length check - I remember reading something about the library not liking `ForceSendFields` if the field would have been sent anyway.  Maybe that was `NullFields`... I'm just going to submit it this way, then go check.  :)

Reply:
It's `NullFields`!  Still, I think I prefer this style a little bit even knowing that the other way would work.  I guess it's not important in the long run since our use of this library is dwindling.  :)